src/main/java/io/cryostat/net/web/http/api/v2/CredentialGetHandler.java

```diff
@@ -0,0 +1,127 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package io.cryostat.net.web.http.api.v2;
+
+import java.io.IOException;
+import java.util.EnumSet;
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import io.cryostat.configuration.CredentialsManager;
+import io.cryostat.configuration.CredentialsManager.MatchedCredentials;
+import io.cryostat.messaging.notifications.NotificationFactory;
+import io.cryostat.net.AuthManager;
+import io.cryostat.net.security.ResourceAction;
+import io.cryostat.net.web.http.HttpMimeType;
+import io.cryostat.net.web.http.api.ApiVersion;
+import io.cryostat.platform.ServiceRef;
+
+import com.google.gson.Gson;
+import io.vertx.core.http.HttpMethod;
+
+class CredentialGetHandler extends AbstractV2RequestHandler<MatchedCredentials> {
+
+    private final CredentialsManager credentialsManager;
+    private final NotificationFactory notificationFactory;
+
+    @Inject
+    CredentialGetHandler(
+            AuthManager auth,
+            CredentialsManager credentialsManager,
+            NotificationFactory notificationFactory,
+            Gson gson) {
+        super(auth, gson);
+        this.credentialsManager = credentialsManager;
+        this.notificationFactory = notificationFactory;
+    }
+
+    @Override
+    public boolean requiresAuthentication() {
+        return true;
+    }
+
+    @Override
+    public ApiVersion apiVersion() {
+        return ApiVersion.V2_2;
+    }
+
+    @Override
+    public HttpMethod httpMethod() {
+        return HttpMethod.GET;
+    }
+
+    @Override
+    public Set<ResourceAction> resourceActions() {
+        return EnumSet.of(ResourceAction.READ_CREDENTIALS);
+    }
+
+    @Override
+    public String path() {
+        return basePath() + "credentials/:id";
+    }
+
+    @Override
+    public HttpMimeType mimeType() {
+        return HttpMimeType.JSON;
+    }
+
+    @Override
+    public boolean isAsync() {
+        return false;
+    }
+
+    @Override
+    public boolean isOrdered() {
+        return true;
+    }
+
+    @Override
+    public IntermediateResponse<MatchedCredentials> handle(RequestParameters params)
+            throws ApiException {
+        int id = Integer.parseInt(params.getPathParams().get("id"));
+        try {
+            String matchExpression = credentialsManager.get(id);
+            Set<ServiceRef> targets = credentialsManager.resolveMatchingTargets(id);
+            MatchedCredentials match = new MatchedCredentials(matchExpression, targets);
+            return new IntermediateResponse<MatchedCredentials>().body(match);
+        } catch (IOException ioe) {
+            throw new ApiException(500, ioe);
+        }
```

Comment:
If I try to get credentials with a nonexistent id, I get back an HTTP 500 instead of a 404. Should the response be a 404 Not Found instead?

```
curl http://0.0.0.0:8181/api/v2.2/credentials/2
{"meta":{"type":"text/plain","status":"Internal Server Error"},"data":{"reason":null}}
```

Reply:
Thanks, fixed. I also added unit tests which I realized were missing for these handlers.